### PR TITLE
GUI2/Rich Label: minor cleanup

### PR DESCRIPTION
**Pull request by Vultraz**
Couple cleanup/optimization thinks I saw looking through the code.

### CODE REVIEW

**Review by babaissarkar — APPROVED** [review]

**Comment by babaissarkar** [comment]
look good, tho out of curiosity, why was the `push_back` -> `emplace_back` needed?

**Comment by Vultraz** [comment]
Slightly more efficient. It just means the object is constructed directly in the list instead of creating a temporary and then moving it into the list.

File: src/gui/widgets/rich_label.cpp [hunk 1]
```diff
@@ -403,26 +375,23 @@ std::pair<config, point> rich_label::get_parsed_text(
...
 				col_idx = 0;
 				max_row_height = 0;
 
-				for (config& col : row.child_range("col")) {
-					config col_cfg;
-					config& col_txt_cfg = col_cfg.add_child("text");
-					col_txt_cfg.append_attributes(col);
-					col_cfg.append_children(col);
+				for(const config& col : row.child_range("col")) {
+					config col_cfg{"text", col};
```
Comment (Vultraz) [inline src/gui/widgets/rich_label.cpp:384]: I just realized I misread the original code here. Only the *attributes* go in the `text` tag... Will fix.

File: src/gui/widgets/rich_label.cpp [hunk 2]
```diff
@@ -72,49 +79,52 @@ rich_label::rich_label(const implementation::builder_rich_label& builder)
 		std::bind(&rich_label::signal_handler_mouse_leave, this, std::placeholders::_3));
 }
 
-wfl::map_formula_callable rich_label::setup_text_renderer(config text_cfg, unsigned width) {
+wfl::map_formula_callable rich_label::setup_text_renderer(config text_cfg, unsigned width) const {
 	// Set up fake render to calculate text position
-	wfl::action_function_symbol_table functions;
+	static wfl::action_function_symbol_table functions;
```
Comment (CelticMinstrel) [inline src/gui/widgets/rich_label.cpp:84]: Is it really okay for this to be static? Feels questionable.

File: src/gui/widgets/rich_label.cpp [hunk 3]
```diff
@@ -72,49 +79,52 @@ rich_label::rich_label(const implementation::builder_rich_label& builder)
...
+	return {
+		variables.query_value("text_width").as_int(),
+		variables.query_value("text_height").as_int()
+	};
 }
 
-point rich_label::get_image_size(config& img_cfg) {
-	wfl::action_function_symbol_table functions;
+point rich_label::get_image_size(config& img_cfg) const {
+	static wfl::action_function_symbol_table functions;
```
Comment (CelticMinstrel) [inline src/gui/widgets/rich_label.cpp:103]: Ditto.

**Pull request merged**